### ISSUE TITLE
img_range parameter incorrect for BasicSR implementation

### ISSUE THREAD

**Issue by terrainer** [hongyuanyu/SPAN#6]
Hi,

Thank you for your code! This performs very impressively at such fast speeds and low memory usage.
This BasicSR implementation has one major flaw in its `img_range` parameter. The default value is `255.`, but a much more suitable value would be `1.`.

This is because BasicSR normalizes inputs to a range of `[0,1]`. EDSR, which is seemingly where the `255.` value comes from, does not do this, and also does its zero-centering via a much different MeanShift function. You'll notice that any other network with a BasicSR implementation and this same mean/range code, [such as SwinIR ](https://github.com/JingyunLiang/SwinIR/blob/6545850fbf8df298df73d81f3e8cba638787c8bd/models/network_swinir.py#L651), uses the value of `1.`.

Currently, the usage of this value vastly blows up expected ranges of `[0,1]` or `[-0.5, 0.5]` to `[-127.5, 127.5]`, causing high instability. Unfortunately, the editing of this parameter also breaks inference for anything expecting `img_range` to be `255.`, and so the current official pretrains would not work if inference were to be done with `img_range: 1.`. They would have to be re-trained with the updated value.

Kind regards,
terrainer

**Comment by reddured** [comment]
Could this issue be related to the problem you mentioned ?
https://github.com/hongyuanyu/SPAN/issues/9

**Comment by reddured** [comment]
I noticed you finetuned some SPAN models yourself (like SPANkendata). Have you managed to correct this issue or at least mitigate pixel artifacts on output image ? It doesn't seem to appear on the comparison shot related to your SPANkendata model (https://github.com/terrainer/AI-Upscaling-Models/blob/main/4xSPANkendata/comparison/Upscale.png). Thanks.

**Comment by terrainer** [comment]
The issue is fixed within chaiNNer and neosr.

**Comment by reddured** [comment]
Do you mean this is fixed in chaiNNer / neosr by using a denoising algorithm on output image (like NLM denoising) ?

**Comment by terrainer** [comment]
ChaiNNer and neosr have changed the code itself to fix the img_range issue.

This also makes models trained using this repos code incompatible with
those two.
